### ISSUE TITLE
Sql tests with external oracle database fail

### ISSUE THREAD

**Issue by JiriOndrusek** [apache/camel-quarkus#3256]
If you run `sql` integration tests with external oracle db, it fails.
Oracle connection paramaters have to be exported like:
```
export SQL_JDBC_USERNAME=...
export SQL_JDBC_PASSWORD=...
export SQL_JDBC_URL=...
```
Command to use external db: `mvn clean verify -Dcq.sqlJdbcKind=oracle -Dnative=true -pl :camel-quarkus-integration-test-sql`

Exception in log:
```
Caused by: java.lang.IllegalStateException: Object serialization is currently not supported
	at java.io.ObjectStreamClass.<init>(ObjectStreamClass.java:66)
	at java.io.ObjectStreamClass.lookup(ObjectStreamClass.java:381)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1135)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
	at org.apache.camel.processor.aggregate.jdbc.JdbcCamelCodec.encode(JdbcCamelCodec.java:99)
```

**Comment by JiriOndrusek** [comment]
First impression is that everything looks like it should work. 
I tried some different scenarios and results show another problem with oracle jdbc.

If I run tests (native) with `h2` (default configuration), results are successful.
If I add `quarkus-jdbc-oracle` into dependencies, jvm tests are successful, but native image build fails.

```
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-jdbc-oracle</artifactId>
        </dependency>
```

Exception is:
```
[camel-quarkus-integration-test-sql-2.5.0-SNAPSHOT-runner:25]     analysis:  50,242.45 ms,  5.67 GB
[camel-quarkus-integration-test-sql-2.5.0-SNAPSHOT-runner:25]      [total]:  58,229.00 ms,  5.67 GB
Error: Classes that should be initialized at run time got initialized during image building:
 java.sql.DriverManager the class was requested to be initialized at run time (from feature io.quarkus.runner.AutoFeature.beforeAnalysis with 'DriverManager.class'). To see why java.sql.DriverManager got initialized use --trace-class-initialization=java.sql.DriverManager

com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 java.sql.DriverManager the class was requested to be initialized at run time (from feature io.quarkus.runner.AutoFeature.beforeAnalysis with 'DriverManager.class'). To see why java.sql.DriverManager got initialized use --trace-class-initialization=java.sql.DriverManager

	at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.checkDelayedInitialization(ConfigurableClassInitialization.java:555)
	at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.duringAnalysis(ClassInitializationFeature.java:169)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$12(NativeImageGenerator.java:730)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:730)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:532)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:491)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:380)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:543)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:119)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:573)
# Printing build artifacts to: /project/camel-quarkus-integration-test-sql-2.5.0-SNAPSHOT-runner.build_artifacts.txt
Error: Image build request failed with exit status 1
```
This exception is the same as in this closed issue - https://github.com/apache/camel-quarkus/issues/2448

Interesting fact is, that if there is only `quarkus-jdbc-oracle` and no `h2`, image build succeeds (failure happens in test execution)

1. From my POV having `quarkus-jdbc-h2` and `quarkus-jdbc-oracle` in one project should work. 
2. We should enable tests of different db types on CI (https://github.com/apache/camel-quarkus/issues/3083) and then we should also created something like `sql-grouped` to be sure that all jdbc drivers work together.
What do you think @ppalaga?

**Comment by ppalaga** [comment]
> * From my POV having `quarkus-jdbc-h2` and `quarkus-jdbc-oracle` in one project should work.

Maybe we should ask Quarkus team whether combining drivers is supported.

> * We should enable tests of different db types on CI ([Sql - enable  all the DB kind tests on the CI #3083](https://github.com/apache/camel-quarkus/issues/3083)) and then we should also created something like `sql-grouped` to be sure that all jdbc drivers work together.

Yep, that would be nice to have if it is supported by Quarkus.

**Comment by JiriOndrusek** [comment]
@ppalaga I was told recently that Oracle isn't supported at Native by Quarkus (Only JVM).

In that case my scenario with `h2` and `oracle` is also not supported at this time.

I plan to dig deeper into Oracle native support and try to find if I can help help with this problem.

**Comment by JiriOndrusek** [comment]
Blocked by https://github.com/quarkusio/quarkus/issues/20396

**Comment by JiriOndrusek** [comment]
Proble, is not happening (verified on 267358ca79fa443e4a3bcf55ef8ef2ec3d98d798), Issue can be closed.
